internal/controllers/core/tiltfile/reconciler.go

```diff
@@ -63,16 +79,178 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	}
 
 	if apierrors.IsNotFound(err) || !tf.ObjectMeta.DeletionTimestamp.IsZero() {
-		// TODO(nick): Handle deletion
+		// TODO(nick): Also delete any owned objects.
+		r.deleteExistingRun(nn)
 		return ctrl.Result{}, nil
 	}
 
 	ctx = store.MustObjectLogHandler(ctx, r.st, &tf)
-	_ = ctx
+	run := r.runs[nn]
+
+	// If the spec has changed, cancel any existing runs
+	if run != nil && !apicmp.DeepEqual(run.spec, &(tf.Spec)) {
+		run.cancel()
+		delete(r.runs, nn)
+		run = nil
+	}
+
+	step := runStepNone
+	if run != nil {
+		step = run.step
+		ctx = run.entry.WithLogger(ctx, r.st)
+	}
+
+	// Check to see if we have a BuildEntry triggering this tiltfile.
+	//
+	// This is a short-term hack to make this interoperate with the legacy subscriber.
+	// The "right" version would read from FileWatch and other dependencies to determine
+	// when to build.
+	be := r.buildSource.Entry()
+	if be != nil && be.Name.String() == nn.Name && (step == runStepNone || step == runStepDone) {
```

Comment:
Why does this trigger on `runStepDone`?

Reply:
like if the tiltfile restarts! will add a comment on this